preparedata/src/main/resources/org/orbisgis/orbisprocess/geoclimate/preparedata/bdtopo/importPreprocess.sql

```diff
@@ -135,6 +146,24 @@ CREATE TABLE $INPUT_VEGET AS SELECT a.the_geom, a.ID as ID_SOURCE, a.NATURE as T
 -- Update the VEGET table with the new appropriate type key, coming from the abstract table
 UPDATE $INPUT_VEGET SET TYPE=(SELECT c.TERM FROM $VEGET_BD_TOPO_TYPE b, $VEGET_ABSTRACT_TYPE c WHERE c.ID_TYPE=b.ID_TYPE and ${INPUT_VEGET}.TYPE=b.NATURE) WHERE TYPE IN (SELECT b.NATURE FROM $VEGET_BD_TOPO_TYPE b);
 
+-------------------------------------
+-- Impervious areas (from the layers "TERRAIN_SPORT", "CONSTRUCTION_SURFACIQUE", "SURFACE_ROUTE" and "SURFACE_ACTIVITE") that are in the study area (ZONE)
+-------------------------------------
+DROP TABLE IF EXISTS $TMP_IMPERV_TERRAIN_SPORT, $TMP_IMPERV_CONSTRUCTION_SURFACIQUE, $TMP_IMPERV_SURFACE_ROUTE, $TMP_IMPERV_SURFACE_ACTIVITE;
+CREATE TABLE $TMP_IMPERV_TERRAIN_SPORT AS SELECT a.the_geom, a.ID as ID_SOURCE FROM $TERRAIN_SPORT a, $ZONE b WHERE a.the_geom && b.the_geom AND ST_INTERSECTS(a.the_geom, b.the_geom) AND (a.NATURE='Piste de sport' OR a.NATURE='Terrain de tennis');
+CREATE TABLE $TMP_IMPERV_CONSTRUCTION_SURFACIQUE AS SELECT a.the_geom, a.ID as ID_SOURCE FROM $CONSTRUCTION_SURFACIQUE a, $ZONE b WHERE a.the_geom && b.the_geom AND ST_INTERSECTS(a.the_geom, b.the_geom) AND (a.NATURE='Barrage' OR a.NATURE='Ecluse' OR a.NATURE='Escalier');
+CREATE TABLE $TMP_IMPERV_SURFACE_ROUTE AS SELECT a.the_geom, a.ID as ID_SOURCE FROM $SURFACE_ROUTE a, $ZONE b WHERE a.the_geom && b.the_geom AND ST_INTERSECTS(a.the_geom, b.the_geom);
+CREATE TABLE $TMP_IMPERV_SURFACE_ACTIVITE AS SELECT a.the_geom, a.ID as ID_SOURCE FROM $SURFACE_ACTIVITE a, $ZONE b WHERE a.the_geom && b.the_geom AND ST_INTERSECTS(a.the_geom, b.the_geom) AND (a.CATEGORIE='Administratif' OR a.CATEGORIE='Enseignement' OR a.CATEGORIE='Santé');
```

Comment:
Good point.
- For `TERRAIN_SPORT` ([see](http://professionnels.ign.fr/doc/DC_BDTOPO_2.pdf#page=104)) : after checking in Dept 56, seems only "Piste de sport" is OK (e.g Kercado stadium is OK while it's only grass in OSM)--> I will remove "Terrain de tennis"
- For `CONSTRUCTION_SURFACIQUE` ([see](http://professionnels.ign.fr/doc/DC_BDTOPO_2.pdf#page=114)) : I guess that by definition 'Barrage' / 'Ecluse' / 'Escalier' are impervious
- For SURFACE_ROUTE ([see](http://professionnels.ign.fr/doc/DC_BDTOPO_2.pdf#page=35)). Def : 'Partie de la chaussée d’une route caractérisée par une largeur exceptionnelle (place, carrefour, péage, parking). Zone à trafic non structuré.' --> so we can consider this is impervious
- For SURFACE_ACTIVITE ([see](http://professionnels.ign.fr/doc/DC_BDTOPO_2.pdf#page=135)) : I already removed many CATEGORIES since there were ambiguous (e.g 'Industriel ou commercial '). IMO 'Administratif', 'Enseignement' and 'Santé' are impervious by definition ?!

Reply:
OK for "pistes de sport", construction surfacique, surface route but I am not sure the surface activite is necessary. I think that they have been excluded from the OSM scheme since it is difficult to know with certainty that they are impervious (for example for "enseignement" it may be more green in the future - while today it is clearly "grey"...).

Reply:
> for example for "enseignement" it may be more green in the future - while today it is clearly "grey"...
Yes but here we are talking about the BD Topo V2.2, which is already outdated ;-)
For the future (v3.0, ...) and other dataset, we can adapt.

I have no fixed opinion.
We'll always have some fake cases, but overall with these 3 cases ('Administratif', 'Enseignement' and 'Santé'), I think we're OK.
@ebocher, what do you think ? We keep `SURFACE_ACTIVITE` or not ?

Reply:
The layer "SURFACE_ACTIVITE" does not guarantee that all geometries represent impervious surfaces. I propose not to keep it because there is a risk of too much overestimation.

Reply:
I changing the code in order to make these layers optional.
If there is no input layers, we create a corresponding empty layers (with the expected columns) and we run the .sql code as is it. So even if we don't have objects, the process runs without crashing.

Reply:
Done. we now check if the 4 impervious tables are here. If one (or more) of them is missing, we create the corresponding empty layer. No modification in the .sql file. Just in the .groovy